hack/lib/golang.sh

```diff
@@ -292,7 +292,9 @@ kube::golang::server_test_targets() {
 IFS=" " read -ra KUBE_TEST_SERVER_TARGETS <<< "$(kube::golang::server_test_targets)"
 readonly KUBE_TEST_SERVER_TARGETS
 readonly KUBE_TEST_SERVER_BINARIES=("${KUBE_TEST_SERVER_TARGETS[@]##*/}")
-readonly KUBE_TEST_SERVER_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]}")
+if [[ -n "${KUBE_SERVER_PLATFORMS:-}" ]]; then
+  readonly KUBE_TEST_SERVER_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]}")
```

Comment:
Instead of ifs, just change this line to:
```suggestion
  readonly KUBE_TEST_SERVER_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]:+"${KUBE_SERVER_PLATFORMS[@]}"}")
```

Reply:
@cblecker  thank you for the suggestion, your version is really nit with just one line, but i find it is less obvious and took me sometime to understand.

i would prefer to keep the current ifs if the extra line make it a little easier to understand, thanks

Reply:
@figo Yours does something different, however. Yours doesn't even set `KUBE_TEST_SERVER_PLATFORMS` because it's an if.

Please make the requested change. Thank you!

Reply:
@cblecker i see what you mean now, PR refreshed, thx